[Python] Type mismatch error for optional parameter

For the following two-file program:
`ex.py`:
```python
import ll
from ll import Foo

f: Foo = ll.create('foo', t='t')
```
`ll.py`:
```python
class Foo:
    def __init__(self, t: str) -> None:
        self.t = t
    

class Bar:
    def __init__(self, t: str) -> None:
        self.t = t
    
def create(
    s: str,
    t: str,
    b: bool | None = None,
) -> Foo | Bar:
    if s == 'foo':
        return Foo(t)
    elif s == 'bar':
        return Bar(t)
    else:
        raise NotImplementedError("Unknown class")
```
ESBMC errors out:
```bash
$ esbmc ex.py 
ESBMC version 7.11.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing ex.py

Type checking warning:
ex.py:4: error: Incompatible types in assignment (expression has type "Foo | Bar", variable has type "Foo")  [assignment]
Found 1 error in 1 file (checked 1 source file)

Converting
WARNING: Union with no recognized types, defaulting to pointer
Generating GOTO Program
GOTO program creation time: 0.580s
GOTO program processing time: 0.002s
Starting Bounded Model Checking
ERROR: function call: argument "py:/home/ubuntu/examples/python/pygo/ErgoPythonBenchmarks/api-error-benchmarks/botomoogified/bug1/ll.py@F@create@b" type mismatch: got pointer, expected struct
Aborted (core dumped)
```
ESBMC version: f2fedb1e5

--- a/regression/python/github_3087/ll.py
+++ b/regression/python/github_3087/ll.py
@@ -1,0 +1,20 @@
+class Foo:
+    def __init__(self, t: str) -> None:
+        self.t = t
+    
+
+class Bar:
+    def __init__(self, t: str) -> None:
+        self.t = t
+    
+def create(
+    s: str,
+    t: str,
+    b: bool | None = None,
+) -> Foo | Bar:
+    if s == 'foo':
+        return Foo(t)
+    elif s == 'bar':
+        return Bar(t)
+    else:
+        raise NotImplementedError("Unknown class")

--- a/regression/python/github_3087/main.py
+++ b/regression/python/github_3087/main.py
@@ -1,0 +1,4 @@
+import ll
+from ll import Foo
+
+f: Foo = ll.create('foo', t='t')

--- a/regression/python/github_3087/test.desc
+++ b/regression/python/github_3087/test.desc
@@ -1,0 +1,4 @@
+CORE
+main.py
+
+^VERIFICATION SUCCESSFUL$

--- a/src/python-frontend/python_converter.cpp
+++ b/src/python-frontend/python_converter.cpp
@@ -1903,14 +1903,41 @@ exprt python_converter::get_function_call(const nlohmann::json &element)
       args[it->second] = arg_expr;
     }
 
-    // Fill empty arguments with None for optional parameters
+    // Fill empty arguments with proper Optional values or None for optional parameters
     for (size_t i = 0; i < args.size(); ++i)
     {
       if (args[i].is_nil() || args[i].id().empty())
       {
-        constant_exprt none_expr(none_type());
-        none_expr.set_value("NULL");
-        args[i] = none_expr;
+        const typet &param_type = params[i].type();
+
+        // Check if this is an Optional type (struct with "is_none" field)
+        if (param_type.is_struct())
+        {
+          const struct_typet &struct_type = to_struct_type(param_type);
+          const std::string &tag = struct_type.tag().as_string();
+
+          if (tag.starts_with("tag-Optional_"))
+          {
+            // Create Optional value with is_none=true
+            constant_exprt none_expr(none_type());
+            none_expr.set_value("NULL");
+            args[i] = wrap_in_optional(none_expr, param_type);
+          }
+          else
+          {
+            // Regular struct type - set to NULL pointer
+            constant_exprt none_expr(none_type());
+            none_expr.set_value("NULL");
+            args[i] = none_expr;
+          }
+        }
+        else
+        {
+          // Non-struct type - use NULL for None
+          constant_exprt none_expr(none_type());
+          none_expr.set_value("NULL");
+          args[i] = none_expr;
+        }
       }
     }
   };

